### PR TITLE
Add the information regarding docs

### PR DESCRIPTION
**Pull request by piyush-garg**
This will add the information regarding generating
and updating docs(markdown and man pages) in the PR

Also updated the docs according to current master

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -76,7 +76,8 @@ While iterating on the project, you may need to:
 
 1. [Install Tekton pipeline](#install-pipeline)
 2. [Building Tekton client](#building-tekton-client)
-1. [Add and run tests](./test/README.md#tests)
+3. [Add and run tests](./test/README.md#tests)
```
Comment (sthaha) [inline DEVELOPMENT.md]: fyi: markdown simplifies ordered list e.g. 

```md
1. foo 
1. bar
1. baz
```

File: docs/cmd/tkn_pipeline_logs.md [hunk 2]
```diff
@@ -15,17 +15,17 @@ Show pipeline logs
 ### Examples
 
 
-  # show logs interactively for no inputs
+  # interactive mode: shows logs of the selected pipeline run
```
Comment (sthaha) [inline docs/cmd/tkn_pipeline_logs.md]: Won't this be rendered as H1? 

# interactive mode: ...

File: docs/cmd/tkn_task_list.md [hunk 3]
```diff
@@ -19,7 +19,6 @@ Lists tasks in a namespace
 ```
       --allow-missing-template-keys   If true, ignore any errors in templates when a field or map key is missing in the template. Only applies to golang and jsonpath output formats. (default true)
   -h, --help                          help for list
-  -l, --limit int                     limit taskruns listed (default: return all taskruns)
```
Comment (sthaha) [inline docs/cmd/tkn_task_list.md]: why is this removed?

File: docs/cmd/tkn_pipeline_logs.md (hunk 2 above)
Comment (piyush-garg) [reply docs/cmd/tkn_pipeline_logs.md]: yes, need to change example in the respective file

File: DEVELOPMENT.md (hunk 1 above)
Comment (piyush-garg) [reply DEVELOPMENT.md]: oh nice, I was not aware of this, so corrected the order.

File: docs/cmd/tkn_pipeline_logs.md (hunk 2 above)
Comment (vdemeester) [reply docs/cmd/tkn_pipeline_logs.md]: Yeah, those need to be updated in the source code (maybe we can remove some and add those in a md file so it only appears on man pages and reference docs)

**Comment by piyush-garg** [comment]
@hrishin PR has been updated with the current master.
